### PR TITLE
cdktf: make terraform test dep and update homepage

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/cdktf.rb [hunk 1]
```diff
@@ -16,8 +16,8 @@ class Cdktf < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "ba5b8e53e095c31190d725d5b1bfc8292d6f73da31da893bd257e6fc71747d45"
   end
 
+  depends_on "terraform" => :test
```
Comment (carlocab) [inline Formula/cdktf.rb:19]: Why?

Comment (chenrui333) [reply Formula/cdktf.rb:19]: no need to install the latest `terraform`

Comment (SMillerDev) [reply Formula/cdktf.rb:19]: But you need terraform to use this tool. It is the literal definition of depending on terraform.

Comment (chenrui333) [reply Formula/cdktf.rb:19]: yeah, but the latest, it would potentially suffer the same issue as terragrunt and etc.

Comment (chenrui333) [reply Formula/cdktf.rb:19]: Some discussion thread in here, https://github.com/orgs/Homebrew/discussions/4037

**Pull request closed without merging**